lmfit/minimizer.py

```diff
@@ -498,6 +530,294 @@ def scalar_minimize(self, method='Nelder-Mead', params=None, **kws):
+
+        if ntemps > 1:
+            # the prior and likelihood function args and kwargs are the same
+            sampler_kwargs['loglargs'] = lnprob_args
+            sampler_kwargs['loglkwargs'] = lnprob_kwargs
+            sampler_kwargs['logpargs'] = (bounds,)
+        else:
+            sampler_kwargs['args'] = lnprob_args
+            sampler_kwargs['kwargs'] = lnprob_kwargs
+
+        # now initialise the samplers
+        if reuse_sampler:
+            if auto_pool is not None:
+                self.sampler.pool = auto_pool
+
+            p0 = self._lastpos
+            if p0.shape[-1] != self.nvarys:
+                raise ValueError("You cannot reuse the sampler if the number"
+                                 "of varying parameters has changed")
+        elif ntemps > 1:
+            # Parallel Tempering
+            # jitter the starting position by scaled Gaussian noise
+            p0 = 1 + np.random.randn(ntemps, nwalkers, self.nvarys) * 1.e-4
+            p0 *= var_arr
+            self.sampler = emcee.PTSampler(ntemps, nwalkers, self.nvarys,
+                                           _lnpost, _lnprior, **sampler_kwargs)
+        else:
+            p0 = 1 + np.random.randn(nwalkers, self.nvarys) * 1.e-4
+            p0 *= var_arr
+            self.sampler = emcee.EnsembleSampler(nwalkers, self.nvarys,
+                                                 _lnpost, **sampler_kwargs)
+
+        # user supplies an initialisation position for the chain
+        # If you try to run the sampler with p0 of a wrong size then you'll get
+        # a ValueError. Note, you can't initialise with a position if you are
+        # reusing the sampler.
+        if pos is not None and not reuse_sampler:
+            tpos = np.asfarray(pos)
+            if p0.shape == tpos.shape:
+                pass
+            # trying to initialise with a previous chain
+            elif (tpos.shape[0::2] == (nwalkers, self.nvarys)):
+                tpos = tpos[:, -1, :]
+            # initialising with a PTsampler chain.
+            elif ntemps > 1 and tpos.ndim == 4:
+                tpos_shape = list(tpos.shape)
+                tpos_shape.pop(2)
+                if tpos_shape == (ntemps, nwalkers, self.nvarys):
+                    tpos = tpos[..., -1, :]
+            else:
+                raise ValueError('pos should have shape (nwalkers, nvarys)'
+                                 'or (ntemps, nwalkers, nvarys) if ntemps > 1')
+            p0 = tpos
+
+        # now do a production run, sampling all the time
+        output = self.sampler.run_mcmc(p0, steps)
+        self._lastpos = output[0]
+
+        # discard the burn samples and thin
+        chain = self.sampler.chain[..., burn::thin, :]
+        lnprobability = self.sampler.lnprobability[:, burn::thin]
+
+        flatchain = chain.reshape((-1, self.nvarys))
+
+        mean = np.mean(flatchain, axis=0)
+        quantiles = np.percentile(flatchain, [15.8, 84.2], axis=0)
```

Comment:
Just a suggestion: maybe return also the 2-sigma quantiles? MCMC is most useful when the estimation from the covariance fails, and in these cases 2-sigma may not be 1-sigma*2. Alternative, take the quantils as a parameter But as the chain is returned anyways, it doesnt matter too much.